### ISSUE TITLE
[$500] Workspace - Invite members to workspace error but user can still join the workspace

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#25592]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open 2 users on 2 browsers
2. B send message to A
3. A create a workspace
4. A invite B and invalid user ([+252 3 234211](tel:+2523234211)), observed an error appearing in 2 users when inviting
5. A go to announce room and send message
6. From B, Observe that B can access the room

## Expected Result:
When the invite user fails, B cannot access the workspace

## Actual Result:
B can access the workspace during invite error

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.55-7

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/971f2b38-d29e-48ec-8b3c-3979ff760643

https://github.com/Expensify/App/assets/78819774/a7ea2029-b25d-42d8-a67d-99eecd853888

**Expensify/Expensify Issue URL:**

**Issue reported by:** @namhihi237

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1691655815906349

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017361e55ce3e1ac2d</li>
        <li>Upwork Job ID: 1693979555238805504</li>
        <li>Last Price Increase: 2024-05-02</li>
<li>Automatic offers: </li>
<ul>
<li>s77rt | Reviewer | 102505066</li>
<li>tienifr | Contributor | 102505069</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @bondydaa</details>

**Comment by WaqasIbrahim** [comment]
# Proposal
### Please re-state the problem that we are trying to solve in this issue.
Invited members to a workspace shows an error but the user can still access the workspace.

### What is the root cause of that problem?
The issue title is misleading, the cause of the problem is not that the user is not added. The user is added successfully but the malformed phone number is the cause of the error.

API Response:
<img width="1102" alt="Screenshot 2023-08-23 at 1 47 33 AM" src="https://github.com/Expensify/App/assets/12554146/d5412add-8e05-4a0d-9d60-fc18e1a3bcaf">

Error message: "The provided phone number does not match the country you are currently located at, please use your email address instead."

This part assumes that the request failed completely for all users.
https://github.com/Expensify/App/blob/0f68829b176458ddb8349bde9921f81805944403/src/libs/actions/Policy.js#L394-L399

### What changes do you think we should make in order to solve the problem?
You should filter out users and only show error message for users that were failed to add, not to all.

**Comment by WaqasIbrahim** [comment]
Contributor details
Your Expensify account email: waqas.abrahim@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~018d936bc5c4987b6f

**Comment by s77rt** [comment]
@WaqasIbrahim Thanks for the proposal. Indeed the other user seem to "partly" get access to the workspace. I don't see them in the members list but according to the OP video they can access restricted rooms. This should be handled internally.

**Comment by s77rt** [comment]
@laurenreidexpensify This seems internal. Can you please apple the label?

**Comment by WaqasIbrahim** [comment]
@s77rt You are welcome. Just a suggestion, UX wise this should be handled the same way internally. I (as a user) should not have to reselect the members and add to workspace if last request failed because of one invalid member.

**Comment by laurenreidexpensify** [comment]
@bondydaa this one is internal, so adding engineering for extra eyes

**Comment by bondydaa** [comment]
Sorry it's not clear to me here in the steps.

A is a validated user with an existing account?
B is a validated user with an existing account but isn't already on the workspace that A owns?

The problem is that if you try to add an existing user and also invite an invalid phone/email the admin gets an error back but B is properly added (as they should be, right?) but the invalid phone/email isn't added, right?

**Comment by bondydaa** [comment]
I see but B should be and is added right?

I think @WaqasIbrahim was right in their analysis here https://github.com/Expensify/App/issues/25592#issuecomment-1688921768

We are assuming _all_ users failed even though it failed only for 1 user (the invalid phone number)

So I think I agree this is mostly just a front end thing that is incorrectly showing B had an error event though it didn't.

Though I guess I'm not sure if the error thrown states which specific user had the error so maybe we do need to possibly update the API to return that, let me try to confirm that real quick.

**Comment by bondydaa** [comment]
hmm I wasn't logged in to my "B" user here initially but after I logged in looks like the room names are there 
<img width="1213" alt="image" src="https://github.com/Expensify/App/assets/4073354/25ba8469-75c0-4058-982e-06ec6ead8d7e">

so I bet the error in the API prevent us from hitting the code that should have pushed all the updates out to the right  clients. 

inspecting the network request it dosen't look like the error has the accountID(s) it failed for 
<img width="548" alt="image" src="https://github.com/Expensify/App/assets/4073354/70f57b8a-0832-4ef0-973a-b62fbada5f91">

We could possibly go through the onyx data for the `POLICYCHANGELOG_ADD_EMPLOYEE` b/c that shows which employee we added 
<img width="543" alt="image" src="https://github.com/Expensify/App/assets/4073354/d622e613-0787-4021-a9a1-d808410b37a5">

but that seems kind of hacky to me, I think the API should probably return better which accountIDs were successful and which had errors.

**Comment by bondydaa** [comment]
wowza I was hoping to make this a first pick but this code is confusing 😭 

I believe this is where we are hitting the spot where we try to create the accounts if they don't exist https://github.com/Expensify/Web-Expensify/blob/d506283338f/lib/PolicyAPI.php#L1096 @Beamanator @neil-marcellini I see your names in the blame around here so maybe you have a better idea here 🙏. 

Can you think of an easy way of capturing which logins we might have failed for and which ones we didn't fail for that we could then return from the API here?

**Comment by bondydaa** [comment]
oh also logs for that request https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A%227fe67c237cc65200-DEN%22%20AND%20timestamp%3A%5B2023-08-29T00%3A00%20TO%202023-08-30T23%3A59%5D&index=_all

**Comment by Beamanator** [comment]
Do you know exactly what is failing when inviting new members?

If you can get it from `$modifyEmployeesResponse` that would be nice, but i haven't messed with `self::modifyEmployees` much, and it's very scary code haha. 

If account creation fails, maybe you can tell from `Auth::getAccountIDsByEmails`? I think that won't return accountIDs for emails that don't exist, and I don't think it'll fail - so you could possibly map out which email you invited but didn't get an accountID?

**Comment by laurenreidexpensify** [comment]
Not overdue @bondydaa is OOO today - American holiday

**Comment by bondydaa** [comment]
> Do you know exactly what is failing when inviting new members?

In this exact scenario the failure is because we block trying to create an account for a phone number that is outside of the country you are in... i know super strange and actually probably something we might need to reconsider since there is a legitimate case where an admin could be adding users from a different country.

But I don't think the exact reason matters here much, there could be 42 different reasons why 1 account in a big list that you try to add doesn't work and it shouldn't mark them all as failed in that case.

**Comment by bondydaa** [comment]
> something we might need to reconsider since there is a legitimate case where an admin could be adding users from a different country.

We have this https://github.com/Expensify/Expensify/issues/262189 already about addressing that problem so not something we need to concern ourselves with here.

This issue should focus on fixing it so that 1 error while adding doesn't mark them all as failing.

**Comment by laurenreidexpensify** [comment]
@bondydaa @s77rt what do we think next steps are/

**Comment by s77rt** [comment]
@laurenreidexpensify I think @bondydaa may answer this better. I don't have access to https://github.com/Expensify/Expensify/issues/262189

**Comment by bondydaa** [comment]
next steps are to fix the API, not sure this needs to be a daily or in /App since this will need to be fix primarily in the API though maybe it does need some front end changes. 

I don't really have the bandwidth right now to focus on it though.

**Comment by laurenreidexpensify** [comment]
Making this a weekly

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Android - The backspace button on the keypad only deletes one digit at a time, not the entire number.

### What is the root cause of that problem?
- Currently, when the long-press event is triggered, it will call ```props.numberPressed(key)``` like bellow 
https://github.com/Expensify/App/blob/120bcbfde6d0ccfac8a63cf11a4aa7f769a048ea/src/components/BigNumberPad.js#L44-L55
- And the numberPressed function does not clear all the input in case the long-press event is triggered, it just remove one digit
https://github.com/Expensify/App/blob/120bcbfde6d0ccfac8a63cf11a4aa7f769a048ea/src/pages/iou/steps/MoneyRequestAmountForm.js#L135-L153

### What changes do you think we should make in order to solve the problem?
- We can add the param called "isLongPress" to the updateAmountNumberPad like bellow
```
 const updateAmountNumberPad = useCallback(
        (key, isLongPress) => {
            if (shouldUpdateSelection && !textInput.current.isFocused()) {
                textInput.current.focus();
            }
            // Backspace button is pressed
            if (key === '<' || key === 'Backspace') {
                if (currentAmount.length > 0) {
                    if (isLongPress) {
                        setNewAmount('');
                        return;
                    }
                    const selectionStart = selection.start === selection.end ? selection.start - 1 : selection.start;
                    const newAmount = `${currentAmount.substring(0, selectionStart)}${currentAmount.substring(selection.end)}`;
                    setNewAmount(newAmount);
                }
                return;
            }
            const newAmount = MoneyRequestUtils.addLeadingZero(`${currentAmount.substring(0, selection.start)}${key}${currentAmount.substring(selection.end)}`);
            setNewAmount(newAmount);
        },
        [currentAmount, selection, shouldUpdateSelection],
    );
```
- Then when the long-press event is triggered, the handleLongPress will become
```
   const handleLongPress = (key) => {
        if (key !== '<') {
            return;
        }

        props.longPressHandlerStateChanged(true);

        const newTimer = setInterval(() => {
            props.numberPressed(key, true);
        }, 100);
        setTimer(newTimer);
    };
```
### What alternative solutions did you explore? (Optional)

NA

### Result
[Screencast from 22-08-2023 10:36:16.webm](https://github.com/Expensify/App/assets/141406735/3ffa563a-4878-4fd4-b4c4-98084cf973c6)

**Comment by laurenreidexpensify** [comment]
This is still fine as a weekly

**Comment by laurenreidexpensify** [comment]
@bondydaa any capacity to check this out yet?

**Comment by bondydaa** [comment]
sorry no not really, need to focus on finishing off the recarding doc so mostly heads down on that.

**Comment by bondydaa** [comment]
haven't got back into this yet, other projects still taking priority atm.

**Comment by bondydaa** [comment]
still haven't had time to dig back in here

**Comment by bondydaa** [comment]
still haven't gotten back to this, I'm going to demote it to  monthly since it's been months and nothing bad has happened. 

I agree we should still fix this at some point but I have other priorities that need to get wrapped up before EOY. I'll bring it up on the resync engineering threads to see if someone can take it over though as well.

**Comment by bondydaa** [comment]
posted this here https://expensify.slack.com/archives/CC7NECV4L/p1702496004198239?thread_ts=1702417170.002649&cid=CC7NECV4L trying to see if someone else can take it over

**Comment by bondydaa** [comment]
trying to see what wave this might fall under to see if we can hand it off b/c I'm still focused on getting USPM out the door.

**Comment by bondydaa** [comment]
posted here in wave8 https://expensify.slack.com/archives/C036QM0SLJK/p1707841028741169

**Comment by bondydaa** [comment]
looking into this again today and retested it locally and I think something has slightly changed, now it seems none of the members are added to the workspace if a single one errors out which I guess is slightly better than what was happening before but still not great IMO.

https://github.com/Expensify/App/assets/4073354/f5d31b2b-76b2-4664-875a-061ee0126bd5

But I'm going to test out a change locally to see if I can make this better so that users that can be added are and we only then show RBR errors for users who could not be added

**Comment by bondydaa** [comment]
i'm like halfway through figuring this out but hit some small blockers locally. I'll keep plugging on away on this tomorrow 🙏

**Comment by bondydaa** [comment]
I've got a draft PR up here for the API fixes  https://github.com/Expensify/Web-Expensify/pull/41268

asking for help on how I can now send on to the clients/onyx the values for the users who had errors, I've got this mostly working so far 🎉

**Comment by bondydaa** [comment]
okay my PRs for the API changes have been fully deployed so I think we can have this go back to a contributor to finish off the new dot changes. 

What I've done on the API side is update the response a bit so we need to update the optimistic/failure/success data a bit on the client side and better handle the response.

Previously if you'd tried to add more than 1 user and any user failed we would throw an error which would trigger the failure data to get rendered which resulted in the OP bug where then some users were actually added the workspace even though we showed the `failureData` for them.

Now if any user was successfully added to the workspace we won't throw an error at all so the response back to client will be as if everything worked properly. If a user in the list did fail though then the response will have the `errors` key that we can still pick up and RBR to show.

here's an example of the response data where 1 member was added and 1 member failed ![](https://private-user-images.githubusercontent.com/4073354/317445976-6199f225-f142-43ac-b6dd-c23c24bbc0e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1ODEwMzIsIm5iZiI6MTcxMTU4MDczMiwicGF0aCI6Ii80MDczMzU0LzMxNzQ0NTk3Ni02MTk5ZjIyNS1mMTQyLTQzYWMtYjZkZC1jMjNjMjRiYmMwZTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjdUMjMwNTMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGRmNWU2M2RkNDRkM2ExYjNhMGEzMGI4MmFmYzViYjZmMmIzNGU1NWY1MmZkOTZkOTU5ZTQxZTk3N2RhNjg0MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qiX8Oa9mQBmlsI3w1G30Qo1uw3HF-ECuJ-xrYQGHaCU)

I've also updated the API to expect the request to have look for `accountID` instead of just `email` here https://github.com/Expensify/App/blob/e542ba7c2d9e216c2cd401de0de8ed225fff4a7b/src/libs/actions/Policy.ts#L1344-L1348

so that we can then properly return any `optimisticAccountID` that failed to get created back to the clients directly from the request with the errors to render them in the UI.

Let me know if there's anything else that needs clarifying but open for proposals on how to update this.

**Comment by s77rt** [comment]
Not overdue. Looking for proposals

**Comment by laurenreidexpensify** [comment]
Not overdue, looking for proposals

**Comment by laurenreidexpensify** [comment]
Not overdue, looking for proposals

**Comment by bondydaa** [comment]
I don't think we need to wait for the rest of the backend changes to go out here, there's other higher priority things on my plate so not sure how quickly I'll be able to get those out either. 

I think we can make these changes to new dot incrementally anyways so that we can update it now to work with the existing API response as well as the idealized one we've suggested in slack.

I believe the current problem that is preventing us from rendering the updated errors from the API when we get them is that we just set `value` to `null` here https://github.com/Expensify/App/blob/ee5ab95f2c893738fb48756a9002e9d9dadd9a31/src/libs/actions/Policy.ts#L1321 https://github.com/Expensify/App/blob/ee5ab95f2c893738fb48756a9002e9d9dadd9a31/src/libs/actions/Policy.ts#L1328

We need to having something "smarter" that looks at the response and if there are `accountIDs` with `errors` keys we need to render them with RBRs, like how the `failureData` works.

The `failureData` then too needs to be updated to check for `accountIDs` with `errors` and render the message instead of the default that is set optimistically.

**Comment by s77rt** [comment]
@shubham1206agra Thanks for the proposal. Your RCA is correct. Regarding the solution we shouldn't clear the errors for all the accounts as some may have failed to get added.

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by laurenreidexpensify** [comment]
Still waiting on proposals

**Comment by laurenreidexpensify** [comment]
Updated bounty to see if we can find any proposals

**Comment by laurenreidexpensify** [comment]
Catching up with @bondydaa - we think this job's price at $500 is probably more accurate, as we can largely apply this recommended solution here - https://github.com/Expensify/App/issues/25592#issuecomment-2032969494 that has already been suggested

**Comment by s77rt** [comment]
@tienifr Thanks for the proposal. Regarding the RCA, the code did change, now we are not removing any member, we are just clearing their pendingAction and errors.
https://github.com/Expensify/App/blob/228e7696d98817bc6052a52abd39dbc06fbda926/src/libs/actions/Policy.ts#L1320-L1340

To prevent the errors from being cleared we should move the errors clearing logic to `optimisticMembersState` (that's the correct place).

Now onyx is holding the expected data, but still the user is not being rendered, can you look into that?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by laurenreidexpensify** [comment]
Given not reproducible, am closing

**Comment by s77rt** [comment]
@laurenreidexpensify We still have unfinished work here as outlined in @tienifr's video we are missing rendering errors for users that were failed to be invited.

**Comment by s77rt** [comment]
@tienifr Thanks for looking into this. Here are my thoughts:

> we need to have something like successDataPreprocessor

Let's try avoid adding a new pattern i.e. `successDataProcessors`. At first I thought we can use `API.makeRequestWithSideEffects` to wait on the server's response but I see that this function execute the request immediately without being added to the queue.

> we need to keep the personal details data of the optimistic user if that user failed to be added

Makes sense but can we achieve that without that new pattern?

I also noticed that `getPersonalDetailsOnyxDataForOptimisticUsers` is buggy (race condition) where in the `finallyData` we will always clear the personal details of optimistic account even if we "generated/guessed" the correct account id. Maybe we can get rid of this and find a better way to prevent duplicate entries?

**Comment by s77rt** [comment]
The second alternative solution does not look that bad to me. If we assume that a user would have an account id A and the BE returns a different id (id B), I'd assume it's pretty okay that the BE tell us that id A is not correct i.e. clear personal details for account with id A.

**Comment by bondydaa** [comment]
I'm pretty sure this needs to be open still.

I would advise we stop suggesting further back end changes for now, I don't exactly have the time to focus on those at the moment and I believe we can get this working better with the existing response from the API.

**Comment by s77rt** [comment]
@tienifr I'd prefer to not add a new pattern. Can we instead get rid of the part that clears the optimistic personal details? and fix [the duplicate entries](https://github.com/Expensify/App/blob/3ba42e684f75f143d6e7a85e62d23b7d561d7155/src/libs/PersonalDetailsUtils.ts#L160-L163) in another way?

**Comment by laurenreidexpensify** [comment]
@tienifr do you have any revised proposals? thanks

**Comment by laurenreidexpensify** [comment]
@tienifr any updates? thanks

**Comment by s77rt** [comment]
@tienifr Can we extend the successData type to `() => OnyxUpdate[] | OnyxUpdate[]`. If it's a function we pass the server response. 

This is a new pattern too. Usually adding a new pattern just for one case is not the best but we can discuss this in Slack especially if the implementation is not complex

**Comment by s77rt** [comment]
@tienifr You are right. Unfortunately I don't see a clear solution. Having a request success yet contains error is a new pattern too so perhaps we do need something different. I will raise a slack thread to discuss this

**Comment by s77rt** [comment]
@tienifr It's clear but I don't like the fact that we have to define the processor outside `addMembersToWorkspace`. When debugging you won't know that there is other logic outside

**Comment by s77rt** [comment]
@tienifr That's slightly better. Can you please update your proposal with that

**Comment by s77rt** [comment]
@tienifr Thanks! Let's go with that.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/25592#issuecomment-2049501262)

cc @bondydaa

> We need to having something "smarter" that looks at the response

Since having the BE return a success code while having errors is a new pattern we'd need a new mechanism in the App to deal with that specifically the look into the response part.

Alternatively, we can avoid that new pattern in the BE by considering a request a success only if all the members are invited which is the previous behaviour except that the BE should validate all users then invite them all, instead of validate then invite one by one.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by bondydaa** [comment]
i'll review that later today

**Comment by bondydaa** [comment]
sorry haven't gotten to this, busy week with fires and such will dive in again next week.

**Comment by bondydaa** [comment]
still haven't gotten back to this yes, will do so by EOW

**Comment by bondydaa** [comment]
sorry jumped into our recent fires so didn't get to this again last week

**Comment by bondydaa** [comment]
We have an internal issue here https://github.com/Expensify/Expensify/issues/386300 to refactor the AddMembersToWorkspace API which should address some of the bigger backend headaches still lingering in this. 

but I finally looked at @tienifr 's proposal and lets do it as well to get this working for now with how the API functions. 

as we do more 1:1:1 refactors we can continue to improve.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week)

**Comment by s77rt** [comment]
@tienifr Thanks for the PR. I just retested and noticed that the employees list is now keyed by email instead of account id. This gives us a potential easy solution. In [`WorkspaceMembersPage`](https://github.com/Expensify/App/blob/0cb2dc052f01b6a024ac872ad8bf628159fe7873/src/pages/workspace/WorkspaceMembersPage.tsx) if we don't have the employee's personal details, instead of returning early [here](https://github.com/Expensify/App/blob/0cb2dc052f01b6a024ac872ad8bf628159fe7873/src/pages/workspace/WorkspaceMembersPage.tsx#L320-L323), we create a fallback `details` object. That's almost all we have to do. Please test from your end and update your proposal.

**Comment by s77rt** [comment]
@bondydaa A new solution came to light and also a new BE bug 😅 it should be easy to fix though, the `AddMembersToWorkspace` response contains 2 onyx updates responsible for the employees list, this should be combined into one. Notice in the below screenshot that the second onyx update is targeting the correct object `policy_ID.employeeList` while first is targeting directly `policy_ID` which results in the error not being applied correctly.

<img width="488" alt="Screenshot 2024-06-01 at 10 11 11 PM" src="https://github.com/Expensify/App/assets/16493223/5f398fbd-ae96-4bee-a5f9-2264d15e0031">

**Comment by s77rt** [comment]
@tienifr That's a BE bug https://github.com/Expensify/App/issues/25592#issuecomment-2143611119

**Comment by laurenreidexpensify** [comment]
@miljakljajic handing this one over for parental leave cover. next steps here are waiting on @bondydaa to address comments above. Thanks!

**Comment by bondydaa** [comment]
i'm working on https://github.com/Expensify/Expensify/issues/386300 (internal issue) which is to refactor AddMemebersToWorkspace (again) to be 1:1:1 and hopefully make it not suck again so once that's done we can circle back here to make front end up dates. once i've wrapped my head around what needs to change for 1:1:1 i'll look at the comments requested above for backend changes and see how they can work together.

**Comment by miljakljajic** [comment]
@bondydaa is out this week but the update on https://github.com/Expensify/Expensify/issues/386300 says he's nearly done there. Shouldn't be long til we can review this one again.

**Comment by s77rt** [comment]
Let's reopen this

**Comment by miljakljajic** [comment]
@bondydaa let us know your thoughts on this one when you're back from OOO

**Comment by s77rt** [comment]
Melvin is looking to be OOO too 😆

**Comment by miljakljajic** [comment]
I've DM'd @bondydaa and he'll provide an update shortly. In the meanwhile, I'm going to re-assign this as my parental leave begins next week.

**Comment by bondydaa** [comment]
Sorry been focused on other thing but we have been trying to improve the backend performance of this API endpoint on this internal issue https://github.com/Expensify/Expensify/issues/386300

The main thing we were trying to do is remove the need for the front end to deal with the sharing out of data after a user is added so that instead that data is pushed out to clients by the server.

I _think_ we still need to better address the error handling on the front end but will need to double check that to be sure.

**Comment by zanyrenney** [comment]
This has been open nearly 18 months. We are not prioritising expense right now to the highest degree as we are the other channels. I also can't reproduce this when I just tried! I'm not sure if there's a specific flow you need to take. But I'm gonna close this as it's not a priority right now.
